File: src/0001.two-sum.1/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    // hash缓存已访问的数据
    const map = {};
    for (let i = 0; i < nums.length; i++) {
        if (map[target - nums[i]] !== undefined) {
            return [
                map[target - nums[i]], i, ];
        }
        map[nums[i]] = i;
    }
};

File: src/0001.two-sum.1/solution.go
package twosum1

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		index, ok := m[target-nums[i]]
		if ok {
			return []int{index, i}
		}
		m[nums[i]] = i
	}
	return []int{-1, -1}
}
